File: src/main/kotlin/acc/model/DocumentType.java
package acc.model;

import acc.util.Messages;

public enum DocumentType {
    INVOICE(Messages.Faktura.cm()),
    BANK_STATEMENT(Messages.Vypis_z_uctu.cm()),
    ELSE(Messages.Ostatni.cm());

    private DocumentType(String text) {
        this.text = text;
    }

    private String text;

    /**
     * @return the text
     */
    public String getText() {
        return text;
    }
}

File: src/main/kotlin/acc/model/DocumentType.kt
package acc.model

import acc.util.Messages

enum class DocumentType constructor(
        val text: String) {
    INVOICE(Messages.Faktura.cm()),
    BANK_STATEMENT(Messages.Vypis_z_uctu.cm()),
    INCOME(Messages.Prijmovy_doklad.cm()),
    OUTCOME(Messages.Vydajovy_doklad.cm()),
    EVENT(Messages.Ucetni_udalost.cm())
}
